### PR TITLE
HPCC-26934 Add support for load balancer CIDR ranges on external services

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by richardkchapman** [comment]
@jhiscs FYI

**Comment by richardkchapman** [comment]
This is a copy of the changes submitted by Josh Hiscocks in https://github.com/hpcc-systems/helm-chart/pull/11

File: helm/hpcc/templates/_helpers.tpl [hunk 1]
```diff
@@ -991,6 +993,12 @@ spec:
   selector:
     server: {{ .selector | quote }}
   type: {{ $lvars.type }}
+{{- if $lvars.loadBalancerSourceRanges }}
```
Comment (jakesmith) [inline helm/hpcc/templates/_helpers.tpl:996]: should this also be dependent on $lvars.type ? i.e. don't generate if ClusterIP type..

**Review by jakesmith — COMMENTED** [review]
@richardkchapman - 1 question.

File: helm/hpcc/templates/_helpers.tpl (hunk 1 above)
Comment (richardkchapman) [reply helm/hpcc/templates/_helpers.tpl:996]: Or perhaps give an error. I did wonder, but was not sure whether it's worth the bother.

Comment (jakesmith) [reply helm/hpcc/templates/_helpers.tpl:996]: not sure, I assume it's benign if generated on a ClusterIP service, though it still implies a misconfiguration of sorts.
I think an error is preferable to ignoring.

Comment (richardkchapman) [reply helm/hpcc/templates/_helpers.tpl:996]: Error check added

**Review by ghalliday — APPROVED** [review]
I don't see any problems.

**Review by jakesmith — APPROVED** [review]
@richardkchapman - looks good

**Pull request merged**